src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/ScalarExpressionTransformer.cs

```diff
@@ -0,0 +1,722 @@
+            bool parseSuccessful = true;
+
+            switch (sqlDataTypeOption)
+            {
+                case SqlDataTypeOption.Binary:
+                    paramType = SqlDbType.Binary;
+                    try
+                    {
+                        parsedValue = TryParseBinaryLiteral(literalValue, VariableName, SqlDbType.Binary, literal.StartLine);
+                    }
+                    catch (ParameterizationFormatException)
+                    {
+                        if (IsCodeSenseRequest)
+                        {
+                            parseSuccessful = false;
+                            AddCodeSenseErrorItem(MessageHelper.MessageType.BINARY_LITERAL_PREFIX_MISSING_ERROR, literal, literal.Value, VariableName, SqlDbType.Binary.ToString());
+                        }
+                        else
+                        {
+                            throw;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        parseSuccessful = false;
+                        HandleError(MessageHelper.MessageType.ERROR_MESSAGE, literal, VariableName, SqlDbType.Binary.ToString(), C_SHARP_BYTE_ARRAY, literalValue, literal.StartLine, e);
+                    }
+                    break;
+
+                case SqlDataTypeOption.VarBinary:
+                    paramType = SqlDbType.VarBinary;
+                    try
+                    {
+                        parsedValue = TryParseBinaryLiteral(literalValue, VariableName, SqlDbType.VarBinary, literal.StartLine);
+                        ExtractSize(sqlDataTypeParameters, sqlParameter);
+                    }
+                    catch (ParameterizationFormatException)
+                    {
+                        if (IsCodeSenseRequest)
+                        {
+                            parseSuccessful = false;
+                            string sqlDataTypeString = GetSqlDataTypeStringOneParameter(SqlDbType.VarBinary, sqlDataTypeParameters);
+                            AddCodeSenseErrorItem(MessageHelper.MessageType.BINARY_LITERAL_PREFIX_MISSING_ERROR, literal, literalValue, VariableName, sqlDataTypeString);
+                        }
+                        else
+                        {
+                            throw;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        parseSuccessful = false;
+                        string sqlDataTypeString = GetSqlDataTypeStringOneParameter(SqlDbType.VarBinary, sqlDataTypeParameters);
+                        HandleError(MessageHelper.MessageType.ERROR_MESSAGE, literal, VariableName, sqlDataTypeString, C_SHARP_BYTE_ARRAY, literalValue, literal.StartLine, e);
+                    }
+                    break;
+
+
+                //Integer literals of form 24.0 will not be supported
+                case SqlDataTypeOption.BigInt:
```

Comment:
My gut feeling is that some of these (e.g. BigInt/Int/SmallInt and Money/SmallMoney) can be consolidated.  Maybe worth looking into?

Reply:
I was thinking about whether I should decompose this switch/case to a collection of independent command patterned classes for each datatype that is acquired by some factory method. I debated whether this would be overkill or not?